### PR TITLE
Fixing bug with qconfig check

### PR DESCRIPTION
**Pull request by HDCharles**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #68908

Summary: The assert_valid_qconfig function tries to check whether a
qconfig on a convT model has a per_channel weight qscheme, but the NS
debug qconfig: torch.ao.quantization.QConfig(activation=logger_cls, weight=None)
results in an error when qconfig.weight() would be called. Added a convt
model to NS test to check for this in the future.

Test Plan: python test/test_quantization.py TestEagerModeNumericSuite.test_compare_model_stub_conv_static

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32928449](https://our.internmc.facebook.com/intern/diff/D32928449)

### CODE REVIEW

File: torch/ao/quantization/qconfig.py [hunk 1]
```diff
@@ -249,7 +249,7 @@ def assert_valid_qconfig(qconfig: Optional[Union[QConfig, QConfigDynamic]],
     """
     Verifies that this `qconfig` is valid.
     """
-    if qconfig is None:
+    if qconfig is None or qconfig.weight is None:
```
Comment (vkuzo) [inline torch/ao/quantization/qconfig.py:252]: might be cleaner to move the weight check inside the if statement, something like

```
if is_conv_transpose_mod:
    if qconfig.weight is None:
      # for now, we assume that any qconfig for ConvTranspose without a weight is valid
      return 
    ...
```

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32928449).

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32928449).

**Review by vkuzo — APPROVED** [review]

**Comment by HDCharles** [comment]
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32928449).

**Pull request closed without merging**